### ISSUE TITLE
Is the JAVA EventHubClient Implementation thread-safe?

### ISSUE THREAD

**Issue by adoelger** [Azure/azure-event-hubs#116]
Is the JAVA EventHubClient Implementation thread-safe? I cannot find any documentation about this?
Currenty I'm using one EventHubClient instance with multiple threads and get the follwing Exception:

Caused by: java.lang.NullPointerException
	at com.microsoft.azure.eventhubs.EventHubClient.send(EventHubClient.java:183)
       ....

I'm using Version 0.6.6 of the EventHub Client SDK.

**Comment by SreeramGarlapati** [comment]
Thanks to @bojingo who provided the repro of this issue in #114 :

```
public class App 
{
	private static List<EventData> eventData;

	static class Mod2EventSendSync extends Thread {
        private EventHubClient client;
        private int mod2Value;

        public Mod2EventSendSync(EventHubClient client, int mod2Value) {
            this.client = client;
            this.mod2Value = mod2Value;
        }
        
        @Override
        public void run() {
            for (int i = 0; i < eventData.size(); i++) {
                if (i % 2 == mod2Value) {
                    uncheckedSendSync(client, eventData.get(i));
                }
            }
        }

	}

	private static void uncheckedSendSync(EventHubClient client, EventData data) {
        try {
            client.sendSync(data);
        } catch (ServiceBusException e) {
            throw new RuntimeException(e);
       }
    }


	private static void simpleTwoThreads(final String CONNECTION_STRING) {
        try {
            EventHubClient client = EventHubClient.createFromConnectionStringSync(CONNECTION_STRING);
            LocalDateTime start = LocalDateTime.now();
            System.out.println("Started: " + start);
            Mod2EventSendSync odd = new Mod2EventSendSync(client, 1);
            Mod2EventSendSync even = new Mod2EventSendSync(client, 0);
            odd.start();
            even.start();
            odd.join();
            even.join();
            LocalDateTime stop = LocalDateTime.now();
            System.out.println("Stopped: " + stop + "  -  Elapsed milliseconds: " + ChronoUnit.MILLIS.between(start, stop));
        }
        catch (Exception e) {
            e.printStackTrace();
        }
    }
	
	private static List<EventData> generateData() throws UnsupportedEncodingException {
        List<EventData> eventData = new ArrayList<>();
        for (int i = 0; i < 100; i++) {
            String dataMessage = "Message " + i + " being sent...";
            byte[] data = dataMessage.getBytes("UTF-8");
            eventData.add(new EventData(data));
        }
        return eventData;
    }

	
    public static void main( String[] args ) throws Exception
    {
    	eventData = generateData();

    	String username = "RootManageSharedAccessKey";
		String password =  "iU9ySWuCqSIXpi2VOWCmJ5jgvLke/A+Sy1ksh/omW4o=";
		String namespaceName = "javaClient";
		ConnectionStringBuilder connStr = new ConnectionStringBuilder(namespaceName, "teststream", username, password);

		simpleTwoThreads(connStr.toString());
    	}
}

```

**Comment by SreeramGarlapati** [comment]
@adoelger, @bojingo
fixed the issue & here's the staged repo:

```
<repositories>  
  <repository>  
            <id>azure eventhubs</id>  
            <url>https://oss.sonatype.org/content/repositories/commicrosoftazure-1272/</url>  
      </repository>  
 </repositories>  

      <dependency> 
        <groupId>com.microsoft.azure</groupId> 
        <artifactId>azure-eventhubs</artifactId> 
        <version>0.6.8</version> 
    </dependency> 
```
do give it a try and let me know how it went...

**Comment by SreeramGarlapati** [comment]
This issue was fixed in PR #117 and  released in - version 0.6.8.
